### ISSUE TITLE
Site Logo: Dark transparent site logos are invisible

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#49501]
When you upload an image to a Site Logo block the first time, a square crop of that image is used also as your Site Icon. This is likely going to be somewhat common for white sites with black text logo types, such as this:

<img width="486" alt="Screenshot 2023-03-31 at 09 24 17" src="https://user-images.githubusercontent.com/1204802/229053220-3b5fe001-a844-4fc9-b8d9-6bfa62f655bc.png">

In that situation, the site icon becomes barely visible:

<img width="846" alt="Screenshot 2023-03-31 at 09 26 20" src="https://user-images.githubusercontent.com/1204802/229053332-f6ace5de-bc6d-4ca4-86a9-0be50b505a85.png">

Let's ensure that the site logo always has the site background applied, so that the color the logo has been designed to rely on is applied for contrast:
<img width="406" alt="Screenshot 2023-05-02 at 08 42 01" src="https://user-images.githubusercontent.com/1204802/235597494-f2d53a31-d2a7-4e58-a705-f1e55f400006.png">

Can just apply the background color to the `img` directly.

**Comment by courtneyr-dev** [comment]
Possibly related? https://wordpress.org/support/topic/gutenberg-dosent-work/

**Comment by jasmussen** [comment]
Seems likely, yes. One idea, add a circular dot behind any image, so there's always going to be something visible at least.

**Comment by richtabor** [comment]
>  It would be easy to add a white or light gray background, and that should do it, right?

What if we used the site's background color as the fallback? Probably would be better with the header template part's background even—though I'm not sure how possible that would be.

**Comment by jasmussen** [comment]
> What if we used the site's background color as the fallback? Probably would be better with the header template part's background even—though I'm not sure how possible that would be.

That seems like it would be a great solution.

**Comment by mtias** [comment]
+1 for using the background of the site, in the same way the global styles card is created.

**Comment by talldan** [comment]
Something that might be considered in conjunction with this is the contrast of the favicon when a browser is in dark mode. This was my test of using a logo like the one in the issue description (1st and 3rd tabs are my WordPress site, and yes I do tend to have too many tabs open):
![Screen Shot 2023-05-03 at 4 06 33 pm](https://user-images.githubusercontent.com/677833/235862633-026d3de5-13e4-4fad-9b46-0740be281208.png)

I wonder if there are any good options for helping users avoid that, like a preview of an example browser tab in the block inspector.

**Comment by mtias** [comment]
You could argue that showing the dark icon being illegible in the admin should already be a hint that you could run into problems. I wonder if we could render a notice or something if you upload a logo with transparency and use it for the site icon.

**Comment by aaronrobertshaw** [comment]
I've started looking into this one.

Adding a fallback background or gradient, following the global styles preview card, is pretty straightforward for the icon in the site editor. Things get trickier for the uses of the site icon in the post editor as it currently doesn't have access to the merged global styles. 

Access to global styles data within the post editor is something that has been looked into a few times already (https://github.com/WordPress/gutenberg/pull/47098, https://github.com/WordPress/gutenberg/pull/34178, https://github.com/WordPress/gutenberg/pull/34156, https://github.com/WordPress/gutenberg/pull/46894). Unfortunately, for the moment we still cant access the global styles colors for the post editor's site icon.

Some possible options might be:

1. Proceed with using the global styles background color or gradient for the site editor's icon background
2. Apply an arbitrary icon background color for the site icon in the post editor
3. Leave the post editor's icon as is until we can access global styles in that editor
4. Revisit this once we can access global styles data in both editors

What are everyone's thoughts on the best way forward?

**Comment by mtias** [comment]
I think this is low priority and we can let it sit for a little bit.

**Comment by lukecarbis** [comment]
Is there any issue for just using the default WordPress logo in the menu instead of the site icon? If this were a setting somewhere, this issue could be lower priority. (That would also partially solve for bad logo crops)

I think adding a dark logo with transparent background is quite a common use case, and creates a very noticeable visual bug within a primary interaction. FWIW, I think the problem should be considered as a higher priority.

**Comment by aurooba** [comment]
Noting that I'm going to spin up a PR for this. :)

**Comment by aaronrobertshaw** [comment]
It's worth noting access to the merged global styles data in the post editor is [in the process of being added](https://github.com/WordPress/gutenberg/pull/59929) for other features such as section styling and displaying inherited style values in the UI.

Perhaps, waiting for that to land will make achieving consistency for the site logo in both editors easier?

**Comment by filipanoscampos** [comment]
Noting that this is an issue that WordPress.com Happiness Engomneers see reported while in support, as logos become invisible when the browsers are set to use dark mode.

**Comment by Mayank-Tripathi32** [comment]
I am working on the PR to implement the discussed solution

**Comment by jasmussen** [comment]
Coming back to this one, it still feels relevant to show the site default background color behind the site logo as the simplest fix for this.

But one other option I wanted to mention is that if the black logo with a transparent background is inserted as an image, which it is in the site logo case, then using some simple filter CSS, we can add a stroke, as shown in [this codepen](https://codepen.io/joen/pen/Kwzbaaw).

<img width="134" height="117" alt="Image" src="https://github.com/user-attachments/assets/f1a6b401-e607-42f0-8ec8-4b8d95c22244" />

```
  filter: 
    drop-shadow(1px 0 0 white)
    drop-shadow(-1px 0 0 white)
    drop-shadow(0 1px 0 white)
    drop-shadow(0 -1px 0 white);
```

Until such a time as the actual global styles background color value can easily be used here, this might be a holdover solution.

**Comment by jasmussen** [comment]
Trying a fix here https://github.com/WordPress/gutenberg/pull/73816

**Comment by jasmussen** [comment]
Noting that unfortunately the fix I tried above was not a good fix. [A useful report](https://github.com/WordPress/gutenberg/pull/73816#pullrequestreview-3614460398) clarified that indeed the best solution si to apply the site background color.

**Comment by Mayank-Tripathi32** [comment]
Could we move forward with previous approach in that case? https://github.com/WordPress/gutenberg/pull/69693

**Comment by jasmussen** [comment]
Yes, that is the most promising path forward. I expanded the ping range.
